.github/workflows/release.yaml

```diff
@@ -0,0 +1,155 @@
+name: release
+on:
+  pull_request:
+    paths:
+    - '.github/workflows/release.yaml'
+    - 'dist/*'
+    - 'tools/*'
+  push:
+    branches:
+    - main
+    tags:
+    - "*"
+
+jobs:
+  package:
+    runs-on: ubuntu-latest
+    env:
+      RUST_VERSION: "1.57.0"
+    strategy:
+      fail-fast: false
+      matrix:
+        arch:
+        - cpu: x86_64
+          platform: amd64
+        os:
+        - name: linux
+          distro: debian
+          version: 11
+          codename: bullseye
+          pkg_type: deb
+        - name: linux
+          distro: ubuntu
+          version: 21.04
+          codename: hirsute
+          kernel: linux
+          pkg_type: deb
+        - name: linux
+          distro: centos
+          version: 7
+          codename: 7
+          pkg_type: rpm
+        - name: linux
+          distro: centos
+          version: 8
+          codename: 8
+          pkg_type: rpm
```

Comment:
Not that I'm aware of. I have my own informal rules for deciding things like that, but this initial matrix was just to make sure both package types work, likewise across distros.

CentOS announced last year that it is changing from its previous RHEL-like release process to a more Fedora-like model (which is completely the opposite of why people run CentOS, but it is what it is), which is why 8 is marked EOL even though 7 isn't.

In general though, I would recommend that we support the latest stable release, and the latest LTS release for the distros we care about. We may also be able to ship only Debian releases for all Debian-derivative distros if they install Postgres in the same location, in which case we could support the most commonly encountered versions (Debian 9, 10, and 11, which correspond to Ubuntu 18.04, 20.04, and 21.04). With CentOS it would be 7, RHEL would be 7 and 8, and future CentOS and Fedora would track the latest release (since they both follow a continuous release model). Again though, we could check and see if a single .rpm package could work across all RHEL/Fedora-like distros, and try to simplify our build matrix that way.

As far as supporting arm64 builds, I suggest we only produce those for the latest stable release in each OS, since those are predominately going to be used for development (ARM servers are still very rare in practice). I'd guess the vast majority of such users would be people running Docker on an M1-series laptop. We can use exclusions in the build matrix to avoid trying to produce arm64 versions for all combinations.

Reply:
👍

Reply:
what is the difference between this matrix and TimescaleDB itself?

Reply:
TimescaleDb builds for PostgreSQL 12-14, on the following (x86_64 only):

- Debian 9, 10, 11
- Ubuntu 18.04, 20.04, 21.04, and 21.10
- Centos 7 and 8

I'm not sure why they build a separate package for Ubuntu 21.10.

The matrix in the PR currently targets Debian 11, Ubuntu 21.04, and Centos 7 and 8.